modules/social_features/social_group/modules/social_group_request/src/Controller/GroupRequestController.php

```diff
@@ -0,0 +1,150 @@
+   */
+  protected $messenger;
+
+  /**
+   * The cache tags invalidator.
+   *
+   * @var \Drupal\Core\Cache\CacheTagsInvalidator
+   */
+  protected $cacheTagsInvalidator;
+
+  /**
+   * The controller constructor.
+   */
+  public function __construct(EntityFormBuilderInterface $entity_form_builder, MessengerInterface $messenger, CacheTagsInvalidator $cache_tags_invalidator) {
+    $this->entityFormBuilder = $entity_form_builder;
+    $this->messenger = $messenger;
+    $this->cacheTagsInvalidator = $cache_tags_invalidator;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity.form_builder'),
+      $container->get('messenger'),
+      $container->get('cache_tags.invalidator')
+    );
+  }
+
+  /**
+   * Return the title for approve request confirmation page.
+   */
+  public function getTitleApproveRequest(GroupInterface $group, GroupContentInterface $group_content) {
+    return $this->t('Approve membership request to group @group_title', ['@group_title' => $group->label()]);
+  }
+
+  /**
+   * Return the title for reject request confirmation page.
+   */
+  public function getTitleRejectRequest(GroupInterface $group, GroupContentInterface $group_content) {
+    return $this->t('Reject membership request to group @group_title', ['@group_title' => $group->label()]);
+  }
+
+  /**
+   * Builds the form to create new membership on membership request approve.
+   */
+  public function approveRequest(GroupInterface $group, GroupContentInterface $group_content) {
+    /** @var \Drupal\group\Plugin\GroupContentEnablerInterface $plugin */
+    $plugin = $group->getGroupType()->getContentPlugin('group_membership');
+
+    // Pre-populate a group membership from Membership request.
+    $group_content = GroupContent::create([
+      'type' => $plugin->getContentTypeConfigId(),
+      'gid' => $group->id(),
+      'entity_id' => $group_content->get('entity_id')->getString(),
+    ]);
+
+    $this->cacheTagsInvalidator->invalidateTags(['request-membership:' . $group->id()]);
+
+    return $this->entityFormBuilder->getForm($group_content, 'add');
+  }
+
+  /**
+   * Callback to request membership.
+   */
+  public function requestMembership(GroupInterface $group) {
+    $contentTypeConfigId = $group
+      ->getGroupType()
+      ->getContentPlugin('group_membership_request')
+      ->getContentTypeConfigId();
+
+    $group_content = GroupContent::create([
+      'type' => $contentTypeConfigId,
+      'gid' => $group->id(),
+      'entity_id' => $this->currentUser()->id(),
+      'grequest_status' => GroupMembershipRequest::REQUEST_PENDING,
+    ]);
+    $result = $group_content->save();
+
+    if ($result) {
+      $this->messenger()->addMessage($this->t("Your request is waiting for Group Administrator's approval"));
+    }
+    else {
+      $this->messenger()->addMessage($this->t("Error creating request"), self::TYPE_ERROR);
```

Comment:
```suggestion
      $this->messenger()->addError($this->t('Error creating request'));
```

Reply:
I changed function to addError but when I  change " to ' the code sniffer don't like it: http://prntscr.com/opwxsn

Reply:
I don't know how did you check it but I checked it locally and I didn't see the error here. Also, Travis build didn't found the error (see [here](https://travis-ci.org/goalgorilla/open_social/jobs/569680479#L1268)).